storage/rocksdb/ha_rocksdb.cc

```diff
@@ -8659,9 +8676,14 @@ int ha_rocksdb::check_and_lock_sk(const uint &key_id,
     return row_info.tx->set_status_error(table->in_use, s, kd, m_tbl_def,
                                          m_table_handler);
   }
+  if (total_order_seek) {
+    setup_iterator_bounds(kd, &new_slice);
```

Comment:
I think the setup_iterator_bounds code should go into can_use_bloom_filter() and the latter should be renamed so that it's clear there's a side effect. This makes it clear the iterator bounds needs to be setup if bloom filters are not being used. So when can_use_bloom_filter() is going to return false, then within the function, immediately call setup_iterator_bounds() to setup the upper/lower bounds.